cmd/juju/cloud/listcredentials.go

```diff
@@ -243,15 +235,22 @@ func (c *listCredentialsCommand) Run(ctxt *cmd.Context) error {
 		ctxt.Infof("secrets are not shown in tabular format")
 		c.showSecrets = false
 	}
-	credentials := credentialsMap{ClientOnly: c.ClientOnly}
+	credentials := credentialsMap{ClientOnly: c.Client}
+	cloudMsg := ""
+	if c.cloudName != "" {
+		cloudMsg = fmt.Sprintf("for cloud %q ", c.cloudName)
+	}
+	if err := c.MaybePrompt(ctxt, fmt.Sprintf("list credentials %vfrom", cloudMsg)); err != nil {
```

Comment:
this prompt will be wrong if c.cloudName == "" since cloudMsg will be ""

Reply:
Yes, it is intentional and reads fine.
If cloud name is given it'll read 
```
list credentials for cloud AAAA from [client/controller]
````
If not, it'll read
```
list credentials from [cloud/controller]
```

I've experimented with this quiet a bit to get the wording and spacing correct.